### PR TITLE
json-writer: print `level` key in SARIF output

### PR DESCRIPTION
**Pull request by lzaoral**
... so that users can easily filter results according to their severity.

Fixes: #80

### CODE REVIEW

File: src/json-writer.cc [hunk 1]
```diff
@@ -214,6 +215,17 @@ static void sarifEncodeMsg(PTree *pDst, const std::string& text)
     pDst->put_child("message", msg);
 }
 
+static void sarifEncodeLevel(PTree *result, const std::string &event) {
+    std::string level = "warning";
+
+    if (boost::starts_with(event, "error"))
+        level = "error";
```
Comment (kdudka) [inline src/json-writer.cc:222]: I do not like the fact that each level is typed twice (in the match and in the assignment).  This is prone to typos as the list will grow.  Could we rather implement it along these lines?
```cxx
    std::string level = event;                                                                
                                                                                              
    // cut the [...] suffix from event if present                                             
    size_t pos = event.find('[');
    if (std::string::npos != pos)
        level = event.substr(0U, pos);

    // go through events that denote warning level                                            
    for (const char *str : {"error", "warning", "note"}) {                                    
        if (str == level) {                                                                   
            // encode in the output if matched                                                
            result->put<std::string>("level", level);
            return;                                                                           
        }
    }   
```

**Review by kdudka — CHANGES_REQUESTED** [review]
Thank you for working on it!  Some minor issues inline...

File: src/json-writer.cc (hunk 1 above)
Comment (lzaoral) [reply src/json-writer.cc:222]: That looks perfect, thanks!

Comment (lzaoral) [reply src/json-writer.cc]: Addressed in a3eac76.

File: src/json-writer.cc [hunk 2]
```diff
@@ -214,6 +215,17 @@ static void sarifEncodeMsg(PTree *pDst, const std::string& text)
     pDst->put_child("message", msg);
 }
 
+static void sarifEncodeLevel(PTree *result, const std::string &event) {
+    std::string level = "warning";
```
[the remark this answers is not in the document]
Comment (lzaoral) [reply src/json-writer.cc]: Addressed in a3eac76.

**Review by kdudka — APPROVED** [review]
Thanks for the update!

**Review by jamacku — APPROVED** [review]
This is amazing, Thank you! :partying_face: 

Screenshot from 2022-08-26 21-19-53

File: src/json-writer.cc [hunk 3]
```diff
@@ -214,6 +214,24 @@ static void sarifEncodeMsg(PTree *pDst, const std::string& text)
     pDst->put_child("message", msg);
 }
 
+static void sarifEncodeLevel(PTree *result, const std::string &event) {
```
Comment (lzaoral) [inline src/json-writer.cc:217]: Note to self: fix bracket formatting inconsistency in some future commit

Comment (kdudka) [reply src/json-writer.cc:217]: We might consider using this one-liner in CI to catch this:
```
grep '^static .*) *{$' src/*
```

There is one more inconsistency in `abstarct-parser.cc`:
```
src/abstract-parser.cc:static inline std::unique_ptr<T> make_unique(InStream &input) {
src/json-writer.cc:static void sarifEncodeLevel(PTree *result, const std::string &event) {
```

Comment (lzaoral) [reply src/json-writer.cc:217]: Using a custom `.clang-format` config to enforce code style is would be a better solution.
tracked in https://github.com/csutils/csdiff/issues/86.

Comment (kdudka) [reply src/json-writer.cc:217]: That would mean more work than the proposed one-liner.  But if we have a volunteer to work on this, why not :-)

Comment (lzaoral) [reply src/json-writer.cc:217]: It's not that hard, I've already done it few times.  Additionally, such config file can be then used in all of our projects.

**Pull request merged**